tools/kagami/src/main.rs

```diff
@@ -286,19 +287,33 @@ mod genesis {
         let register_role =
             RegisterBox::new(Role::new(role_id.clone()).add_permission(token.clone()));
 
-        let register_user_metadata_access = RegisterBox::new(
-            Role::new("CONFIG_AND_USER_METADATA_ACCESS".parse()?)
-                .add_permission(CanSetKeyValueInUserMetadata::new(
-                    "alice@wonderland".parse()?,
-                ))
-                .add_permission(CanRemoveKeyValueInUserMetadata::new(
-                    "alice@wonderland".parse()?,
-                ))
-                .add_permission(CanChangeConfigParameters::new("alice@wonderland".parse()?)),
-        );
         let alice_id = <Account as Identifiable>::Id::from_str("alice@wonderland")?;
-
-        let parameter_defaults = vec![
+        let grant_permission = GrantBox::new(token, alice_id.clone());
+        let grant_permission_to_set_parameters = GrantBox::new(
+            PermissionToken::new("can_set_parameters".parse()?),
+            alice_id.clone(),
+        );
+        let register_user_metadata_access = RegisterBox::new(
+            Role::new("USER_METADATA_ACCESS".parse()?)
+                .add_permission(
+                    PermissionToken::new("can_set_key_value_in_user_account".parse()?).with_params(
+                        [(
+                            "account_id".parse()?,
+                            IdBox::AccountId("alice@wonderland".parse()?).into(),
+                        )],
+                    ),
+                )
+                .add_permission(
+                    PermissionToken::new("can_remove_key_value_in_user_account".parse()?)
+                        .with_params([(
+                            "account_id".parse()?,
+                            IdBox::AccountId("alice@wonderland".parse()?).into(),
+                        )]),
+                ),
```

Comment:
will this fail if there is no `can_set_key_value_in_user_account` smart contract token?